include/libzfs.h

```diff
@@ -732,7 +732,7 @@ extern int zfs_unshare_nfs(zfs_handle_t *, const char *);
 extern int zfs_unshare_smb(zfs_handle_t *, const char *);
 extern int zfs_unshareall_nfs(zfs_handle_t *);
 extern int zfs_unshareall_smb(zfs_handle_t *);
-extern int zfs_unshareall_bypath(zfs_handle_t *, const char *);
+extern int zfs_unshareall_bypath(zfs_handle_t *, const char *, const char *);
```

Comment:
Depending on how you see your man page add in '09, this functionality _should_ have been there from the start. So one COULD consider my change a fix to the "broken" API :).

Just let me know one way or the other what you decide in this matter..